org.eclipse.editorconfig.ui/src/main/java/org/eclipse/editorconfig/ui/internal/EditorConfigVisitor.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2017 Nathan Jones, Jackson Bailey
+ *
+ * This file is part of "EditorConfig Eclipse".
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.eclipse.editorconfig.ui.internal;
+
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.core.runtime.preferences.InstanceScope;
+import org.eclipse.editorconfig.core.ConfigProperty;
+import org.eclipse.editorconfig.core.ConfigPropertyVisitor;
+import org.eclipse.editorconfig.core.EditorFileConfig;
+import org.eclipse.editorconfig.core.EndOfLineOption;
+import org.eclipse.editorconfig.core.IndentStyleOption;
+
+public class EditorConfigVisitor implements ConfigPropertyVisitor {
+
+	private final EditorFileConfig editorFileConfig;
+
+	public EditorConfigVisitor(EditorFileConfig editorFileConfig) {
+		this.editorFileConfig = editorFileConfig;
+	}
+
+	private void setPreference(final String prefsNodeName, final String key, final String value) {
+		System.out.println(String.format("Setting preference: %s/%s=%s", prefsNodeName, key, value));
+		final IEclipsePreferences prefs = InstanceScope.INSTANCE.getNode(prefsNodeName);
+		prefs.put(key, value);
+	}
+
+	private void clearPreference(final String prefsNodeName, final String key) {
+		System.out.println(String.format("Clearing preference: %s/%s", prefsNodeName, key));
+		final IEclipsePreferences prefs = InstanceScope.INSTANCE.getNode(prefsNodeName);
+		prefs.remove(key);
+	}
+
+	@Override
+	public void visitIndentStyle(final ConfigProperty<IndentStyleOption> property) {
+		final Boolean spacesForTabs = property.getValue().equals(IndentStyleOption.SPACE);
+		setPreference("org.eclipse.ui.editors", "spacesForTabs", spacesForTabs.toString());
+		setPreference("org.eclipse.jdt.core", "org.eclipse.jdt.core.formatter.tabulation.char", spacesForTabs ? "space" : "tab");
+		setPreference("org.eclipse.ant.ui", "formatter_tab_char", Boolean.toString(!spacesForTabs));
+	}
+
+	@Override
+	public void visitIndentSize(final ConfigProperty<Integer> property) {
+		final String indentSizeString = property.getValue().toString();
+		final Object indentStyle = editorFileConfig.getConfigProperty("INDENT_STYLE").getValue();
```

Comment:
I added a null-check to avoid exceptions. `indentStyle` can still be null but it shouldn't matter.